internal/scheduler/scheduling_algo.go

```diff
@@ -91,16 +101,45 @@ func (l *FairSchedulingAlgo) Schedule(
 	overallSchedulerResult := &SchedulerResult{
 		NodeIdByJobId: make(map[string]string),
 	}
-	for _, executor := range accounting.executors {
+
+	executorsToSchedule := accounting.executors
+	slices.SortStableFunc(executorsToSchedule, func(a, b *schedulerobjects.Executor) bool {
+		return strings.Compare(a.Id, b.Id) < 1
+	})
+
+	schedCtx, cancel := context.WithTimeout(ctx, l.maxSchedulingDuration)
+	defer cancel()
+
+	allExecutorsConsidered := false
+	for i, executor := range executorsToSchedule {
+		if schedCtx.Err() != nil {
```

Comment:
do we record why we exited- e.g. either because we ran out of time or because we scheduled all executors?  If not, how difficult would it be to add?  Possibly this is something we can do when we do prom metrics